wand-watcher.lic

```diff
@@ -0,0 +1,153 @@
+
+  def passive_run
+    loop do
+      if @wand_list == nil || @wand_list.empty?
+        #@wand_list = nil when no wands configured in yaml
+        #@wand_list = empty when all wands are deleted due to problems finding wands
+        message "No wands to check.  Ending script.  Double check your settings."
+        exit
+      end
+      check_timers
+      pause @passive_delay
+    end
+  end
+
+  def check_timers  
+    #if any no_use_scripts are running, loop every 10 seconds checking until not
+    loop do
+      break if !(@no_use_scripts.any? { |name| Script.running?(name) })
+      pause 10
+    end
+    
+    #Loop through the wand timers in the UserVars, and queue up any wands that are ready to go
+    UserVars.wand_watcher_timers.each do |wand,next_use_time|
+      if Time.now > next_use_time
+        @queued_wands << wand
+      end
+    end
+
+    #If you have any queued wands, use them, then reset the list of queued wands
+    if !@queued_wands.empty?
+      use_wands
+      @queued_wands = []
+    end
+  end
+
+  def use_wands
+    #pause scripts to prevent interference
+    scripts_to_unpause = []
+    scripts_to_unpause = smart_pause_all
+    
+    #if both hands are full, store the left hand item in a variable for future use and lower the 
+    #left hand to the feet slot.  
+    #left causes less problems with combat (less likely to be a loaded weapon)
+    #feet slot causes less space/correct container issues
+    temp_left_item = nil
+    if DRC.right_hand && DRC.left_hand
+      temp_left_item = DRC.left_hand
+      bput("lower ground left",/^You lower/)
+    end
+
+    #redo_count prevents infinite loops when getting a failure message on activation
+    redo_count = 0
+
+    @queued_wands.each do |wand|
+      #get wand settings
+      activation_successful_message = @wand_list[wand]['activation message']
+      activation_verb = @wand_list[wand]['activate verb'] || 'tap'
+      container = @wand_list[wand]['container'] 
+      cooldown = @wand_list[wand]['cooldown'] || 30
+      count = @wand_list[wand]['count'] || 2
+
+      #cooldown is stored in minutes, but used in seconds
+      cooldown *= 60
+      #count is stored as number you own, but ordinals is based on 0 index
+      count -= 1
+      
+      #If you fail getting the wand, because you don't have the right number, or it's not in the 
+      #right container, or you simply don't have them, message the user, and remove the wand from
+      #the list of wands checked
+      if !get_item("#{$ORDINALS[count]} #{wand}",container)
+        message "Could not find correct number of wands for #{wand} in #{container}."
+        message "Double check your wand name, count, an container settings." 
+        message "Removing this wand from the list."
+        @wand_list.delete(wand)
+        UserVars.wand_watcher_timers.delete(wand)
+        next
+      end
+
+      case bput("#{activation_verb} my #{wand}",activation_successful_message,@activation_failure_messages)
```

Comment:
Proposing list of rooms where not to try and use the wands, such as banks and vaults or other areas where magical devices are restricted. This detects if the current room prevents use of the wand.
```suggestion
      case bput("#{activation_verb} my #{wand}",activation_successful_message,@activation_failure_messages,@activation_blocked_messages)
```